protocol/metrics/metrics_consumer_manager.go

```diff
@@ -13,6 +13,20 @@ import (
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
+type WsDisconnectReasonEnum int
+
+const (
+	ConsumerDisconnect WsDisconnectReasonEnum = iota
+	ProviderDisconnect WsDisconnectReasonEnum = iota
+	UserDisconnect     WsDisconnectReasonEnum = iota
+)
```

Comment:
1. You can do the ` = iota` once in thins enum, as can be seen for example in `provider_optimizer.go`
2. Please use upper case snake case for this enum
3. Please prefix the enum items with `WS_DISCONNECTION_REASON_`